Client/multiplayer_sa/CMultiplayerSA_DeviceSelection.cpp

```diff
@@ -0,0 +1,157 @@
+                        WideCharToMultiByte(CP_ACP, 0, targetName.monitorFriendlyDeviceName, -1, monitorFriendlyDeviceName,
+                                            static_cast<int>(std::size(monitorFriendlyDeviceName)), nullptr, nullptr);
+
+                        monitorNames.try_emplace(gdiDeviceName, monitorFriendlyDeviceName);
+                    }
+                }
+            }
+        }
+        FreeLibrary(user32Lib);
+    }
+
+    return monitorNames;
+#endif
+}
+
+struct RwSubSystemInfo
+{
+    char name[80];
+};
+
+#define FUNC_rwDeviceSystemRequest 0x7F2AB0
+using rwDeviceSystemRequest = RwSubSystemInfo*(__cdecl*)(void* device, uint32_t requestId, RwSubSystemInfo* pOut, void* pInOut, uint32_t numIn);
+static RwSubSystemInfo* RwEngineGetSubSystemInfo_Hooked(RwSubSystemInfo* subSystemInfo, int32_t subSystemIndex)
+{
+    static auto rwDeviceSystemRequestFunc = (rwDeviceSystemRequest)(FUNC_rwDeviceSystemRequest);
+    auto        devicePointer = *(uint32_t*)(0xC97B24);
+    auto        result = rwDeviceSystemRequestFunc((void*)(devicePointer + 0x10), 14, subSystemInfo, nullptr, subSystemIndex);
+    if (result == nullptr)
+        return nullptr;
+
+    auto pDxDevice = *(IDirect3D9**)0xC97C20;
+    if (pDxDevice == nullptr)
+        return subSystemInfo;
+
+    D3DADAPTER_IDENTIFIER9 identifier;
+    if (FAILED(pDxDevice->GetAdapterIdentifier(subSystemIndex, 0, &identifier)))
+        return subSystemInfo;
+
+    static const auto friendlyNames = GetFriendlyMonitorNamesForDevicePaths();
+
+    // If we can't find the friendly name, either because it doesn't exist or we're on an ancient Windows, fall back to the device name
+    auto it = friendlyNames.find(identifier.DeviceName);
+    if (it != friendlyNames.end())
+    {
+        strncpy_s(subSystemInfo->name, it->second.c_str(), _TRUNCATE);
+    }
+    else
+    {
+        strncpy_s(subSystemInfo->name, identifier.Description, _TRUNCATE);
+    }
+
+    return subSystemInfo;
+}
+
+#define FUNC_DialogFunc 0x745E50
+static INT_PTR CALLBACK CustomDlgProc(HWND window, UINT msg, WPARAM wParam, LPARAM lParam)
+{
+    auto orgDialogFunc = (DLGPROC)FUNC_DialogFunc;
+    if (msg == WM_INITDIALOG)
+    {
+        orgDialogFunc(window, msg, wParam, lParam);
+
+        // Set Icon
+        HMODULE hGameModule = GetModuleHandle(nullptr);
+        SendMessage(window, WM_SETICON, ICON_SMALL, reinterpret_cast<LPARAM>(LoadIcon(hGameModule, MAKEINTRESOURCE(100))));
+
+        // Make the dialog visible in the task bar
+        // https://stackoverflow.com/a/1462811
+        SetWindowLongPtr(window, GWL_EXSTYLE, WS_EX_APPWINDOW);
+        ShowWindow(window, SW_HIDE);
+        ShowWindow(window, SW_SHOW);
+        return FALSE;
+    }
+
+    return orgDialogFunc(window, msg, wParam, lParam);
+}
+
+void CMultiplayerSA::InitHooks_DeviceSelection()
+{
+    MemPut<DLGPROC>(0x746239, (DLGPROC)&CustomDlgProc);
+    HookInstall(0x7F2C30, (DWORD)RwEngineGetSubSystemInfo_Hooked, 6);
```

Comment:
what are those mem addresses?

Reply:
Added clarification comment to the first one, added a define to the second one to match the current approach over the codebase